integrations/aws/client.go

```diff
@@ -86,10 +92,44 @@ var desc = kittehs.Must1(sdktypes.StrictIntegrationFromProto(&sdktypes.Integrati
 	ConnectionUrl: "/aws/connect",
 }))
 
-func New(vars sdkservices.Vars) sdkservices.Integration {
+func New(cvars sdkservices.Vars) sdkservices.Integration {
+	i := &integration{vars: cvars}
 	return sdkintegrations.NewIntegration(
 		desc,
-		sdkmodule.New(initOpts(vars)...),
-		sdkintegrations.WithConnectionConfigFromVars(vars),
+		sdkmodule.New(initOpts(cvars)...),
+		connStatus(i),
+		sdkintegrations.WithConnectionConfigFromVars(cvars),
 	)
 }
+
+// connStatus is an optional connection status check provided by the
+// integration with AutoKitteh. The possible results are "init required"
+// (indicating the connection is not yet usable), "using X" (indicating
+// one of multiple available authentication methods is in use), or
+// "initialized" when only one authentication method is available.
+func connStatus(i *integration) sdkintegrations.OptFn {
+	return sdkintegrations.WithConnectionStatus(func(ctx context.Context, cid sdktypes.ConnectionID) (sdktypes.Status, error) {
+		if !cid.IsValid() {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "init required"), nil
+		}
+
+		vs, err := i.vars.Get(ctx, sdktypes.NewVarScopeID(cid))
+		if err != nil {
+			return sdktypes.InvalidStatus, err
+		}
+
+		at := vs.Get(authType)
+		if !at.IsValid() || at.Value() == "" {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "init required"), nil
+		}
+
+		// Align with:
+		// https://github.com/autokitteh/web-platform/blob/main/src/enums/connections/connectionTypes.enum.ts
+		switch at.Value() {
```

Comment:
See #595:
1. Remove comment
2. Use a const

Unrelated to #595 :
3. If there's only one option by definition, then use `if-else` instead of `switch`
4. Why not a generic `"initialized"` instead of an AWS-only `"awsConfig"`? Then other integrations could reuse this const

Reply:
1. done
2. done

3. The decision to use switch was based on the idea that we're going to add more ways to connect soon.
4. I thought it had to match one of [these](https://github.com/autokitteh/web-platform/blob/main/src/enums/connections/connectionTypes.enum.ts)

Reply:
Done